### ISSUE TITLE
[User] Dependency Installation steps for ubuntu linux

### ISSUE THREAD

**Issue by TheNotary** [ggerganov/llama.cpp#1146]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

When following the provided documentation, I expect to be able to successfully set up and use the software, with clear and complete instructions guiding me through each step of the process. This includes:

1. Instructions for installing any dependencies and prerequisites.
2. A step-by-step guide for configuring the software, including any necessary configuration files, environment variables, or command-line options.
3. Clear and concise examples of how to use the software, demonstrating its main features and functionalities.
4. Troubleshooting tips or common issues that users may encounter during the setup and usage process, along with their corresponding solutions.
5. Reference to a comprehensive API documentation, if applicable, to allow for customization and integration with other projects.
By following the documentation, I should be able to achieve the desired results and utilize the software as intended by the developers.

# Current Behavior

When attempting to follow the provided documentation, I am unable to successfully set up and use the software due to several gaps and inconsistencies in the instructions. The current issues I am facing missing information on installing dependencies and prerequisites, making it difficult to determine which versions or packages are required for the software to function correctly.

# Environment and Context

OS: Ubuntu 20.04 (running on Windows 11 through WSL2)

# Current Efforts on reconstructing dependency installation steps

```
sudo apt-get install make cmake build-essentials python3 pip

make clean
make
python3 -m pip install -r requirements.txt

cd ..
git clone https://huggingface.co/chavinlo/gpt4-x-alpaca
```

**Comment by CRD716** [comment]
What errors are you getting when you try to build?

**Comment by TheNotary** [comment]
### Error Message

When attempting to build the project by following the instructions provided in the documentation, I encountered the following error:

```
bash: make: command not found
```

This error indicates that the make command is not installed on my system, even though it is required to build the project as per the instructions. The documentation did not mention the need to install make or provide any guidance on how to do so.

**Comment by CRD716** [comment]
`sudo apt install make`

Honestly, if you don't know how to install something in Linux, I would suggest just using cmake on windows or downloading a pre-compiled build for windows.

**Comment by TheNotary** [comment]
Thank you for your suggestion. I understand that using cmake on Windows or downloading a pre-compiled build for Windows could be a potential workaround for my current issue. However, as a user following the project documentation, I expect the instructions to be complete and accurate for the target platform, which in this case is Ubuntu.

Since the project documentation specifically targets Ubuntu users and provides CLI commands for dependency installation, it would be helpful to include the necessary steps for installing make or any other required tools on Ubuntu as well. This will allow users to have a seamless experience when setting up and using the software, regardless of their familiarity with installing software on Linux.

For those who might be interested in using the software on Windows directly, it would be a great addition to the documentation to include instructions on how to build the project with cmake on Windows, or to provide a link to pre-compiled builds for Windows users.

**Comment by yukiteruamano** [comment]
No need to be rude, you have been answered and the answer is correct @TheNotary 

For more info, I have been able to successfully install Dalai Llama both on Docker and without Docker following the procedure described (on Debian) without problems. 

Alpaca and Llama weights are downloaded as indicated in the documentation.

**Comment by syl-00110111** [comment]
i add "git-lfs" package for the downloading of models; it's needed to make git manage large files

**Comment by SlyEcho** [comment]
@syl-00110111 yes, it's very easy to clone HF data with git-lfs. However do note that it stores a second copy of the data in the `.git` directory which can be a significant waste of space with the sizes of files these models have.
Something like this can also be used:
```pwsh
curl.exe -C - -O -L 'https://huggingface.co/chavinlo/gpt4-x-alpaca/resolve/main/pytorch_model-[00001-00006]-of-00006.bin'
```
(Yes, Windows has curl.exe now)

**Comment by TheNotary** [comment]
That's great advice, thanks for sharing that practice, Sly.  

Regarding the rude behavior, I'm sure it's well intentioned but it's easy for confusion to seep into all this GH issues stuff.  Your feelings are important to me.  

> No need to be rude, you have been answered and the answer is correct @TheNotary
> 
> For more info, I have been able to successfully install Dalai Llama both on Docker and without Docker following the procedure described (on Debian) without problems.
> 
> Alpaca and Llama weights are downloaded as indicated in the documentation.
> I apologize if my previous responses seemed to deviate from the main purpose of this issue. My intention is to help improve the README documentation to make it easier for future users to set up and use the project.
>
> Based on the information provided, it seems that you have been able to successfully install and use the project on your system. That's great to hear! However, the current documentation might still benefit from some additional details to ensure a smooth experience for all users, especially those who may not be as familiar with the installation process on Linux.
>
> To address this issue constructively,

I'm willing to lend time towards reflowing the getting started aspect of the readme in a PR from the perspective of a fresh consumer.  If you'd like any direct support there please let me know and thanks for sharing this code @ggerganov, it's pretty amazing stuff 👍
